src/peft/tuners/lora/layer.py

```diff
@@ -150,6 +153,30 @@ def reset_lora_parameters(self, adapter_name, init_lora_weights):
             nn.init.zeros_(self.lora_embedding_A[adapter_name])
             nn.init.normal_(self.lora_embedding_B[adapter_name])
 
+    def olora_init(self, adapter_name):
+        dtype = self.base_layer.weight.dtype
+        if dtype in [torch.int8, torch.uint8]:
+            warnings.warn("You are using OLoRA with quantized weights. This is still experimental.")
```

Comment:
I simply put that to remind that I did not run any benchmarks with quantization yet. We can remove it. its ok to use quantization IMO.

Reply:
Yeah, let's remove it, unless we have a strong suspicion that this will change for the user in the past. The more warnings there are, the more users tend to ignore them.

To be very sure that this works, consider adding GPU tests in `test_gpu_examples.py` using OLoRA. You can copy one of the existing tests and adjust the `LoraConfig`.